projects/step-frontend/src/lib/modules/timeseries/performance-view/chart-generators/chart-generators.ts

```diff
@@ -0,0 +1,99 @@
+import { FindBucketsRequest } from '../../find-buckets-request';
+import { TimeSeriesUtils } from '../../time-series-utils';
+import { UPlotUtils } from '../../uplot/uPlot.utils';
+import { TimeSeriesChartResponse } from '../../time-series-chart-response';
+import { TSChartSettings } from '../../chart/model/ts-chart-settings';
+import { TsChartType } from '../ts-chart-type';
+
+declare const uPlot: any;
+
+export class ChartGenerators {
+  private static readonly CHART_LEGEND_SIZE = 65;
+
+  private static readonly barsFunction = uPlot.paths.bars; // this is a function from uplot which allows to draw bars instead of straight lines
+  private static readonly stepped = uPlot.paths.stepped; // this is a function from uplot wich allows to draw 'stepped' or 'stairs like' lines
+
+  static generateChart(
+    type: TsChartType,
+    request: FindBucketsRequest,
+    response: TimeSeriesChartResponse
+  ): TSChartSettings {
+    switch (type) {
+      case TsChartType.OVERVIEW:
+        return this.createSummaryChartSettings(request, response);
+      case TsChartType.BY_STATUS:
+      case TsChartType.RESPONSE_TIME:
+      case TsChartType.THROUGHPUT:
+      case TsChartType.THREAD_GROUP:
+        throw 'Not implemented';
+    }
+  }
+
+  static createSummaryChartSettings(request: FindBucketsRequest, response: TimeSeriesChartResponse): TSChartSettings {
+    let xLabels = TimeSeriesUtils.createTimeLabels(response.start, response.end, response.interval);
+    let avgValues: (number | null)[] = [];
+    let countValues: (number | null)[] = [];
+    if (response.matrixKeys.length === 0) {
+      // empty data
+      // if (this.summaryChart) {
+      //   this.summaryChart.clear();
+      //   return;
+      // }
+    } else {
+      response.matrix[0].forEach((bucket) => {
+        avgValues.push(bucket ? Math.trunc(bucket.sum / bucket.count) : null);
+        countValues.push(bucket?.throughputPerHour);
+      });
+    }
+
+    return {
+      title: 'Performance Overview',
+      showLegend: true,
+      xValues: xLabels,
+      yScaleUnit: 'ms',
+      series: [
+        {
+          id: 'avg',
+          scale: 'y',
+          label: 'Response Time',
+          data: avgValues,
+          value: (x, v) => Math.trunc(v) + ' ms',
+          // stroke: 'rgba(0,117,187,0.41)',
+          // fill: (self, idx) => {
+          //   let gradient = self.ctx.createLinearGradient(0, 0, 0, 400);
+          //   gradient.addColorStop(0, 'rgba(49,116,197,0.73)');
+          //   gradient.addColorStop(1, '#ff000006');
+          //   return gradient;
+          // },
+          width: 2,
+          stroke: 'rgba(255,109,18,0.59)',
+        },
+        {
+          id: 'count',
+          scale: 'total',
+          label: 'Hits/h',
+          data: countValues,
```

Comment:
I assume that the variable refer to tph and not count.

Reply:
yes. the other is avg, this is the volume/count/tph